test/Microsoft.ML.Tests/Scenarios/Api/Estimators/PredictAndMetadata.cs

```diff
@@ -67,5 +67,68 @@ void PredictAndMetadata()
                 Assert.True(deciphieredLabel == input.Label);
             }
         }
+
+        [Fact]
+        void MulticlassConfusionMatrixSlotNames()
+        {
+            var mlContext = new MLContext(seed: 1);
+
+            var dataPath = GetDataPath(TestDatasets.irisData.trainFilename);
+            var data = mlContext.Data.LoadFromTextFile<IrisData>(dataPath, separatorChar: ',');
+
+            var pipeline = mlContext.Transforms.Concatenate("Features", "SepalLength", "SepalWidth", "PetalLength", "PetalWidth")
+                .Append(mlContext.Transforms.Conversion.MapValueToKey("Label"))
+                .Append(mlContext.MulticlassClassification.Trainers.SdcaMaximumEntropy(
+                    new SdcaMaximumEntropyMulticlassTrainer.Options { MaximumNumberOfIterations = 100, Shuffle = true, NumberOfThreads = 1, }));
+
+            var model = pipeline.Fit(data);
+
+            // Evaluate the model.
+            var scoredData = model.Transform(data);
+            var metrics = mlContext.MulticlassClassification.Evaluate(scoredData);
+
+            // Check that the SlotNames column is there. 
+            Assert.NotNull(scoredData.Schema["Score"].Annotations.Schema.GetColumnOrNull(AnnotationUtils.Kinds.SlotNames));
```

Comment:
>SlotNames [](start = 111, length = 9)

SlotNames would be present only if your label class originally was string. We have TrainingLabelValues which would be available always!

Reply:
it will always be there for the Count column of the confusion matrix. This is what the test is testing, that even when the SlotNames is missing in the Score column, the Count column of the confusion matrix still has the names:


see my comment here: [https://github.com/dotnet/machinelearning/pull/3250#discussion_r274744013](https://github.com/dotnet/machinelearning/pull/3250#discussion_r274744013)

---
In reply to: [275083136](https://github.com/dotnet/machinelearning/pull/3250#discussion_r275083136) [](ancestors = 275083136)